### PR TITLE
Add support for plain .git template URLs

### PR DESCRIPTION
**Pull request by pmuens**
This adds support for plain `.git` URLs pointing to Serverless templates, bypassing the need to use one of our whitelisted code hosting platforms.

You can test it by checking out this PR and running:

```
serverless create --template-url https://github.com/pmuens/serverless-crud.git
```

or

```
serverless create --template-url https://github.com/pmuens/serverless-crud.git --name crud
```

There are no tests yet as I want to see if it fits into the overall picture. @medikoo Just let me know and I'll add some tests.

/cc @tinafangkunding

### CODE REVIEW

File: lib/utils/downloadTemplateFromRepo.js [hunk 1]
```diff
@@ -46,6 +47,14 @@ function validateUrl({ url, hostname, service, owner, repo }) {
   }
 }
 
+/**
+ * Check if the URL is pointing to a Git repository
+ * @param {String} url
+ */
+function isPlainGitURL(url) {
+  return url.endsWith('.git');
```
Comment (medikoo) [inline lib/utils/downloadTemplateFromRepo.js]: Nothing significant but we may double ensure it's `https` (afaik git repo urls can only work via https)

File: package.json [hunk 2]
```diff
@@ -172,6 +172,7 @@
     "filesize": "^3.6.1",
     "fs-extra": "^0.30.0",
     "get-stdin": "^5.0.1",
+    "git-clone": "^0.1.0",
```
Comment (medikoo) [inline package.json]: All it does, is spawning `git` comand.

I think it'll be better to rely on generic `child-process-ext/spawn` which we already use and which provides promise support and eventual debug logging out of a box

**Comment by pmuens** [comment]
@medikoo thanks for the review. Good suggestion all around.

Will push some updates in a minute...

File: package.json (hunk 2 above)
Comment (pmuens) [reply package.json]: Good idea.

**Comment by pmuens** [comment]
@medikoo just addressed the PR comments and added the necessary tests. The PR should be good for a final review...

**Comment by pmuens** [comment]
> CI fails

Apparently the commit messages we not formatted correctly. I've updated it so that they adhere to our requirements.

**Comment by pmuens** [comment]
Thanks for the review @medikoo 👍

**Pull request merged**